internal/standby/standby.go

```diff
@@ -75,11 +87,70 @@ func (s *Service) RunMQTT(ctx context.Context) error {
 			}
 		})
 
+	s.subscribeToTopic(s.cfg.MQTT.ReadCommandTopic,
+		func(client mqtt.Client, msg mqtt.Message) {
+			s.logger.Debug(fmt.Sprintf("Received message: %s from topic: %s", msg.Payload(), msg.Topic()))
+			s.setCommandTimestamp()
+		})
+
 	return nil
 }
 
 func (s *Service) StopMQTT() {
-	s.MQTTClient.Disconnect(uint(1000))
+	s.mqttClient.Disconnect(uint(1000))
+}
+
+func (s *Service) RunDetector(ctx context.Context) {
+	checkInterval := s.cfg.Standby.CheckInterval
+
+	// initialise this so we don't immediately go into failure mode on startup
+	s.setCommandTimestamp()
+
+	ticker := time.NewTicker(checkInterval)
+	for {
+		select {
+		case <-ticker.C:
+			s.CheckForOutage()
+		case <-ctx.Done():
+			ticker.Stop()
+
+			s.logger.Info("Shutting down detector")
+
+			return
+		}
+	}
+}
+
+func (s *Service) CheckForOutage() {
+	outageThreshold := s.cfg.Standby.OutageThreshold
+	timeSinceLastCmd := time.Since(s.getCommandTimestamp())
+	s.logger.Debug("checking", "time since last command", timeSinceLastCmd)
+	if timeSinceLastCmd > outageThreshold {
```

Comment:
let's aim at covering all the branches on this condiontals with tests please

Reply:
all covered now :) https://coveralls.io/jobs/143018627/source_files/26297597053#L107